### PR TITLE
Added Swift example for Key Management

### PR DESCRIPTION
**Pull request by Roy0Anonymous**

### CODE REVIEW

File: docs/key_management.md [hunk 1]
```diff
@@ -43,6 +43,22 @@ val keys_manager = KeysManager.of(
 ```
 
   </template>
+
+  <template v-slot:swift>
+
+```swift
+let seed = [UInt8](repeating: 0, count: 32)
+let timestampSeconds = UInt64(NSDate().timeIntervalSince1970)
+let timestampNanos = UInt32.init(truncating: NSNumber(value: timestampSeconds * 1000 * 1000))
```
[the remark this answers is not in the document]
Comment (Roy0Anonymous) [reply docs/key_management.md]: Updated

File: docs/key_management.md [hunk 2]
```diff
@@ -43,6 +43,22 @@ val keys_manager = KeysManager.of(
 ```
 
   </template>
+
+  <template v-slot:swift>
+
+```swift
+let seed = [UInt8](repeating: 0, count: 32)
```
[the remark this answers is not in the document]
Comment (Roy0Anonymous) [reply docs/key_management.md:50]: Added

File: docs/key_management.md [hunk 3]
```diff
@@ -274,4 +314,133 @@ where
...
 
   </template>
+
+  <template v-slot:swift>
+
+```swift
+class MyKeysManager {
+    let inner: KeysManager
+    let wallet: BitcoinDevKit.Wallet
+	let signerProvider: MySignerProvider
```
[the remark this answers is not in the document]
Comment (Roy0Anonymous) [reply docs/key_management.md]: Updated. I made changes through VSCode, I guess that's what created the indentation issues.

File: docs/key_management.md [hunk 4]
```diff
@@ -274,4 +314,133 @@ where
...
+    }
+    
+    override func getShutdownScriptpubkey() -> Bindings.Result_ShutdownScriptNoneZ {
+        do {
+            let address = try myKeysManager!.wallet.getAddress(addressIndex: .new).address
+            let payload = address.payload()
+            if case let .witnessProgram(`version`, `program`) = payload {
+                let ver: UInt8
+                switch version {
+                case .v0:
```
[the remark this answers is not in the document]
Comment (Roy0Anonymous) [reply docs/key_management.md:380]: I don't think so, this is how XCode indents switch case statements.

File: docs/key_management.md [hunk 5]
```diff
@@ -274,4 +314,133 @@ where
...
+class MySignerProvider: SignerProvider {
+    weak var myKeysManager: MyKeysManager?
+    
+    // We return the destination and shutdown scripts derived by the BDK wallet.
+    override func getDestinationScript() -> Bindings.Result_ScriptNoneZ {
+        do {
+            let address = try myKeysManager!.wallet.getAddress(addressIndex: .new)
+            return Bindings.Result_ScriptNoneZ.initWithOk(o: address.address.scriptPubkey().toBytes())
+        } catch {
+            return myKeysManager!.inner.asSignerProvider().getDestinationScript()
```
[the remark this answers is not in the document]
Comment (Roy0Anonymous) [reply docs/key_management.md]: Thanks, Fixed.

File: docs/key_management.md [hunk 6]
```diff
@@ -43,6 +43,23 @@ val keys_manager = KeysManager.of(
 ```
 
   </template>
+
+  <template v-slot:swift>
+
+```swift
+let seed = [UInt8](repeating: 0, count: 32)
+// Fill in random_32_bytes with secure random data, or, on restart, reload the seed from disk.
```
[the remark this answers is not in the document]
Comment (Roy0Anonymous) [reply docs/key_management.md]: Fixed

**Review by ConorOkus — APPROVED** [review]

**Pull request merged**